x/simulation/params.go

```diff
@@ -105,3 +111,14 @@ func NewSimParamChange(subspace, key, subkey string, simVal SimValFn) ParamChang
 func (spc ParamChange) ComposedKey() string {
 	return fmt.Sprintf("%s/%s/%s", spc.Subspace, spc.Key, spc.Subkey)
 }
+
+//-----------------------------------------------------------------------------
+// Proposal Contents
+
+// WeightedProposalContent defines a common struct for proposal contents defined by
+// external modules (i.e outside gov)
+type WeightedProposalContent struct {
```

Comment:
hmm I'd like to understand a bit more about this. So this is used to created non-`gov` proposals simulation events and their associated simulation weights? This is a bit confusing I almost think this code should somehow exist in the `gov` simulation definitions, as if you're using the simulator without gov this struct / associated functionality would be useless correct?

Reply:
Okay - I understand your thinking now. The only other thing I can think of is to make `WeightedProposalContent` a more generic abstraction interface like `WeightedChange` and keep it in `simulation`. -> then gov would just be fulfilling this interface. 

It's your call whether or not you want to implement this suggested feature - I think it could be good if you feel like giving it a shot. Otherwise that doesn't need to block this PR

Reply:
Okay - I understand your thinking now. The only other thing I can think of is to make WeightedProposalContent a more generic abstraction interface like WeightedChange and keep it in simulation. -> then gov would just be fulfilling this interface.

It's your call whether or not you want to implement this suggested feature - I think it could be good if you feel like giving it a shot. Otherwise that doesn't need to block this PR